kms_log/main.tf

```diff
@@ -558,4 +593,469 @@ resource "aws_cloudwatch_metric_alarm" "dead_letter" {
     alarm_actions = [
         "${var.sns_topic_dead_letter_arn}"
     ]
+}
+
+#lambda functions
+resource "aws_lambda_function" "cloudtrail_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    s3_bucket = "${data.aws_s3_bucket.lambda.id}"
+    s3_key = "circleci/identity-lambda-functions/${var.lambda_identity_lambda_functions_gitrev}.zip"
+
+    lifecycle {
+        ignore_changes = ["s3_key", "last_modified"]
+    }
+
+    function_name = "${var.env_name}-kmslog-ct-processor"
+    description = "18F/identity-lambda-functions: KMS CT Log Processor"
+    role = "${aws_iam_role.cloudtrail_processor.arn}"
+    handler = "main.Functions::KMSCloudTrailHandler.process"  #TODO Verify
+    runtime = "ruby2.5"
+    timeout = 30 # seconds
+
+    environment {
+        variables = {
+            DEBUG = "${var.lambda_audit_github_debug ? "1" : ""}"
+            LOG_LEVEL = "0"
+            CT_SQS_QUEUE = "${aws_sqs_queue.kms_ct_events.id}"
+            DDB_RETENTION_DAYS = "${var.dynamodb_retention_days}"
+            DDB_TABLE = "${aws_dynamodb_table.kms_events.id}"
+            SNS_EVENT_TOPIC_ARN = "${aws_sns_topic.kms_logging_events.arn}"
+        }
+    }
+
+    tags {
+        source_repo = "https://github.com/18F/identity-lambda-functions"
+        environment = "${var.env_name}"
+    }
+}
+
+resource "aws_lambda_event_source_mapping" "cloudtrail_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    event_source_arn = "${aws_sqs_queue.kms_ct_events.arn}"
+    function_name = "${aws_lambda_function.cloudtrail_processor.arn}"
+}
+
+
+data "aws_iam_policy_document" "ctprocessor_cloudwatch" {
```

Comment:
This name is a little confusing because it mentions ctprocessor but it actually applies to cwprocessor too. Could we call it lambda-write-logs or something more accurate like that?

Reply:
The same below for any policy with "ctprocessor" in its name that isn't just for the ctprocessor.

Reply:
Yes it is confusing.  I have updated the names.